cloud/amazon/ecs_taskdefinition.py

```diff
@@ -120,14 +122,95 @@ def __init__(self, module):
+            self.module.fail_json(msg="Can't authorize connection - "+str(e))
 
     def describe_task(self, task_name):
         try:
+            families_response = self.ecs.list_task_definition_families()
+            if task_name not in families_response['families']:
+                return None
+
             response = self.ecs.describe_task_definition(taskDefinition=task_name)
             return response['taskDefinition']
-        except botocore.exceptions.ClientError:
-            return None
+        except botocore.exceptions.ClientError, e:
+            self.module.fail_json(msg="Can't describe task - "+str(e))
+
+    def is_matching_task(self, expected, existing):
+        # prepare containers for comaprison
+        existingContainers = list(existing['containerDefinitions']) if 'containerDefinitions' in existing else []
+        expectedContainers = list(expected['containers']) if 'containers' in expected else []
+
+        for containers in [existingContainers, expectedContainers]:
+            containers.sort(key=lambda k: k['name'])
+            for container in containers:
+                # in case some properties are not defined, we put the default values
+                # and sort lists properties for comaprison
+
+                prop_defaults = {
+                    'disableNetworking': False,
+                    'privileged': False,
+                    'readonlyRootFilesystem': False
+                }
+
+                for prop_name, prop_default in prop_defaults.iteritems():
+                    if prop_name not in container:
+                        container[prop_name] = prop_default
+
+                simple_list_props = [
+                    'links',
+                    'command',
+                    'entryPoint',
+                    'dnsServers',
+                    'dnsSearchDomains',
+                    'dockerSecurityOptions']
+
+                for list_prop in simple_list_props:
+                    if list_prop not in container:
+                        container[list_prop] = []
+                    else:
+                        container[list_prop].sort()
+
+                complex_list_props = {
+                    'environment': 'name',
+                    'mountPoints': 'sourceVolume',
+                    'portMappings': 'containerPort',
+                    'volumesFrom': 'sourceContainer',
+                    'ulimits': 'name',
+                    'extraHosts': 'hostname' }
+
+                for list_prop, sort_key in complex_list_props.iteritems():
+                    if list_prop not in container:
+                        container[list_prop] = []
+                    else:
+                        container[list_prop].sort(key=lambda k: k[sort_key])
+
+                complex_list_defaults = {
+                    'mountPoints': {
+                        'readOnly': False
+                    },
+                    'portMappings': {
+                        'protocol': 'tcp'
```

Comment:
When using dynamic ports, the existing values are returned as:
```
"portMappings": [{
				"containerPort": 80,
				"hostPort": 0,
				"protocol": "tcp"
			}],
```

Should the `complex_list_defaults` entry for `portMappings` be:
```
                  'portMappings': {
                        'protocol': 'tcp',
                        'hostPort': 0
                    },
```